src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java

```diff
@@ -173,6 +182,123 @@ private static String getDate(long date) {
         return simpleDateFormat.format(new java.util.Date(date));
     }
 
+    @Test
+    public void testShortFormAttribute() throws Exception {
+        File testFile = FileSystemTestUtils.getTestFile("org/jnode/fs/xfs/extended_attr.img");
+        try (FileDevice device = new FileDevice(testFile, "r")) {
+            XfsFileSystemType type = new XfsFileSystemType();
+            XfsFileSystem fs = type.create(device, true);
+            final INode shortAttributeINode = fs.getINode(11075L);
```

Comment:
I'm trying to integrate these attribute changes into Engine, so that it can refer to the attributes. I'm guessing the call will be something like `XfsEntry.getINode().getAttributes()` but I don't seem to be able to get that to work. Can you provide an example where you're retrieving the attributes by either the `XfsEntry` or `XfsFile` or `XfsDirectory`?

Reply:
Made it so that tests use `XfsEntry`

Reply:
I was thinking something like this:
```
    @Test
    public void testAttributes() throws Exception {
        File testFile = FileSystemTestUtils.getTestFile("org/jnode/fs/xfs/extended_attr.img");
        try (FileDevice device = new FileDevice(testFile, "r")) {
            XfsFileSystemType type = new XfsFileSystemType();
            XfsFileSystem fs = type.create(device, true);

            XfsEntry rootEntry = fs.getRootEntry();
            FSDirectory directory = rootEntry.getDirectory();
            Iterator<? extends FSEntry> iterator = directory.iterator();
            StringBuilder sb = new StringBuilder("\n");

            while(iterator.hasNext()) {
                XfsEntry entry = (XfsEntry) iterator.next();
                sb.append(entry.getName());
                sb.append("; ");
                sb.append(entry.getAttributes().size());
                sb.append('\n');
            }

            System.out.println(sb);
        } finally {
            testFile.delete();
        }
    }
```
I don't think we're ever going to use `XfsFileSystem.getINode()` - we will most likely walk the tree from the root.
The test above has the following output:
```
.; 0
..; 0
short-form-attr.txt; 1
leaf-attr.txt; 31
node-attr.txt; 201
attr.sh; 1
sparse.dat; 1
```
which at least shows that the attribute list from the files is at least returning something. I'll need to debug more from the Nuix Engine side.

Reply:
I could extract the attribute from the file, so I guess this is resolved. We can probably get away from having the public `XfsFileSystem.getINode()` function as it may not have much public use?